libminifi/include/utils/RegexUtils.h

```diff
@@ -150,18 +132,32 @@ class Regex {
   int regex_mode_;
 #endif
 
-  friend bool regexMatch(const std::string &string, const Regex& regex);
-  friend bool regexMatch(const std::string &string, SMatch& match, const Regex& regex);
-  friend bool regexSearch(const std::string &string, const Regex& regex);
-  friend bool regexSearch(const std::string &string, SMatch& match, const Regex& regex);
+  friend bool regexMatch(const char* str, CMatch& match, const Regex& regex);
+  friend bool regexSearch(const char* str, CMatch& match, const Regex& regex);
+
+  friend bool regexMatch(const std::string_view& str, SVMatch& match, const Regex& regex);
+  friend bool regexSearch(const std::string_view& str, SVMatch& match, const Regex& regex);
+
+  friend bool regexMatch(const std::string& str, SMatch& match, const Regex& regex);
+  friend bool regexSearch(const std::string& str, SMatch& match, const Regex& regex);
+
   friend SMatch getLastRegexMatch(const std::string& string, const utils::Regex& regex);
 };
 
-bool regexMatch(const std::string &string, const Regex& regex);
-bool regexMatch(const std::string &string, SMatch& match, const Regex& regex);
+bool regexMatch(const char* str, const Regex& regex);
+bool regexMatch(const char* str, CMatch& match, const Regex& regex);
+bool regexSearch(const char* str, const Regex& regex);
+bool regexSearch(const char* str, CMatch& match, const Regex& regex);
+
+bool regexMatch(const std::string_view& str, const Regex& regex);
+bool regexMatch(const std::string_view& str, SVMatch& match, const Regex& regex);
+bool regexSearch(const std::string_view& str, const Regex& regex);
+bool regexSearch(const std::string_view& str, SVMatch& match, const Regex& regex);
```

Comment:
Can you add some unit tests for these new overloads, please?

Reply:
added tests